go/client/logclient.go

```diff
@@ -201,10 +216,12 @@ func (c *LogClient) addChainWithRetry(path string, chain []ct.ASN1Cert) (*ct.Sig
 				break
 			case httpResp.StatusCode == 408:
 			case httpResp.StatusCode == 503:
+				fallthrough
+			case err != nil:
```

Comment:
Can this ever happen if the `if err != nil` check was false at L210?
Maybe you meant to remove that `if() {} else {}` entirely?